docs/source/libmav/libmav.py

```diff
@@ -0,0 +1,1046 @@
+    
+    Messages can be recieved either synchronously (blocking) or asychronously.
+    Note again that the messages can come from any MAVLink node on the channel, and there is no inbuilt filtering to just get messages from a particular component or system.
+    
+    Applications register for asynchronous callbacks using the `add_message_callback()` method, passing a callback function and optional exception handler, and de-register using `remove_message_callback()`.
+    The callback is called for every message recieved over the connection.
+    
+    Applications synchronously recieve a specific message using either of the two `receive()` methods (the methods block until the indicated message is received, or there is a timeout):
+    - The method that takes an "expectation" should be used when the expected message is sent in response to another message, such as a `COMMAND_ACK` in response to a particular `COMMAND_INT` message (the "expectation" is created using the `expect()` method).
+    - The method that takes a `message` argument should be used when the desired message is being streamed.
+
+    Lastly, you can syncronously check if the connection is alive using the `alive()` method.
+    If don't want to poll, you can get get asynchronous notification when a connection is lost using `NetworkRuntime.on_connection_lost()`.
+    """
+
+    def add_message_callback(self, callbackFn):
+        """Add a callback function that is called whenever any message is received on the connection.
+
+        The callback function is called with a `Message` object.
+        For example, to print the name of every message, you might use the following code.
+
+        ```python
+        callback_handle = connection.add_message_callback(lambda msg: print(f'{msg.name}'))
+        ```
+
+        A callback can be removed using `remove_message_callback()`
+
+        Args: 
+            callbackFn (function): A callback function that takes a `Message` argument. This is called for every received message.
+        """
+
+
+    def add_message_callback(self, callbackFn, exceptionFn):
+        """Add a callback function for new messages with an exception handler callback.
+        
+        The callback function is called with a `Message` object.
+        The exception handler callback is called with an `Exception` if an exception is raised in the callbackFunction. 
+        <!-- raised from the network? - what errors do we get? -->
+
+        <!-- Example code? -->
+
+        A callback can be removed using `remove_message_callback()`
+
+        Args: 
+            callbackFn (function): A callback function that takes a `Message` argument. This is called for every received message.
+            exceptionFn (function): A callback function that takes an `Exception` argument. This is called if there is an exception in the message callback <!-- correct? -->
```

Comment:
Is there an example we might add of this?

Reply:
Hmm, an easy example would be computing message rates, for example. Always update the number of messages in a dict `{[message_name]: count}`

Reply:
Sounds reasonable. But what I meant to ask was "how does the  exceptionFn work"? I.e. when will you get these errors?Does this end the connection?